### ISSUE TITLE
Code Coverage from Test Task not working with Onsite Build Agent but fine on hosted

### ISSUE THREAD

**Issue by johnkattenhorn** [Microsoft/vsts-tasks#3313]
The Test Task doesn't add any code coverage if we use are on-premise build agent. I think this might be because we were missing Visual Studio Professional but I've tried installing this and it's still not producing the coverage attachment or appearing in the UI.

Running the same build using Hosted Agent works fine. I've checked the agent installation documentation but I've not missed anything there. 

Does anyone know what I've missed or configuration I need to do on our build machines ?

**Comment by DavidStaheli** [comment]
Hi @johnkattenhorn,

Do you know which version of on-premises TFS you are using, or which version of the build agent is running on-premises?

Since the build agent is also available independently, you could upgrade to the latest version [here](https://github.com/Microsoft/vsts-agent).

**Comment by johnkattenhorn** [comment]
Hi @DavidStaheli 

We are using On-premise build agents connected to our Visual Studio Team Services subscription.

I ran a update agent against the pool and the Agent.Capability reads `1.105.6`, does this help track our issue ?

Thanks

John

**Comment by johnkattenhorn** [comment]
I've just upgraded the Agent so that the Agent.Capability reads `2.110.0`, unfortunately the problem still remains.

**Comment by johnkattenhorn** [comment]
This is now resolved but I thought I'd put the resolution here in case it helps anyone else:

I found I needed to:

Install Visual Studio 2015 Test Agents into the build server. I believe I could have also installed Visual Studio 2015 Enterprise as the Code Coverage components are included too but the Test Agent made more sense.

xUnit Preview Test Runner - Once the Test Agent bits were installed we found that our Test Task started to stall at the discovery phase. We currently use xUnit, we were on the latest bits but there seemed to be a lot more commits being released into the preview packages so we upgraded as a test and everything started working.
